### DISCUSSION TITLE
negative cluster label

### DISCUSSION THREAD

**Discussion by FrenchSauce** [parashardhapola/scarf#79] [General]
Hi!

I noticed that there is a '-1' cluster label generated after clustering (both with Leiden and Paris) it doesn't contain much cells (4 out of 50k in the test I run), I was just wondering whether it is a bug or something expected, I haven't found any mention of this in the documentation

`clusters = ds_sample.cells.fetch_all('RNA_leiden_cluster')
np.unique(clusters)`

output:
`array([-1,  1,  2,  3,  4,  5])`

Thanks!

**Comment by parashardhapola** [comment]
Hi @FrenchSauce

`fetch_all` returns the data for all the cells, even those that were filtered out. Since the filtered out cells were not used for clustering they were assigned a value of -1. To ignore the data of filtered out cells you can use `fetch` instead of `fetch_all`. 

Hope this helps.
